### ISSUE TITLE
`empty_docs` triggered in proc-macros

### ISSUE THREAD

**Issue by daxpedda** [rust-lang/rust-clippy#12377]
### Summary

The just newly introduced #12342 is triggered in proc-macros.

### Lint Name

empty_docs

### Reproducer

I tried this code:

```rust
#[wasm_bindgen]
extern "C" {
	type Test;
}
```

I saw this happen:

```
warning: empty doc comment
   --> tests/util/web.rs:141:1
    |
141 | #[wasm_bindgen]
    | ^^^^^^^^^^^^^^^
    |
    = help: consider removing or filling it
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#empty_docs
    = note: this warning originates in the attribute macro `wasm_bindgen` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I expected to see this happen:
The lint not to trigger.

### Version

```text
rustc 1.76.0 (07dca489a 2024-02-04)
binary: rustc
commit-hash: 07dca489ac2d933c78d3c5158e3f43beefeb02ce
commit-date: 2024-02-04
host: x86_64-unknown-linux-gnu
release: 1.76.0
LLVM version: 17.0.6
```

### Additional Labels

@rustbot label +T-macros

**Comment by ilyagr** [comment]
I'm still getting this problem, even on the new beta release.

They show up in cli/src/template_parser.rs and lib/src/revset.rs of https://github.com/martinvonz/jj/tree/f865c1bc5d96f609d2919d559de045dda8256c7c.


rustc 1.78.0-beta.1 (efd9d2df1 2024-03-18)
binary: rustc
commit-hash: efd9d2df12b5e17fac0b4d0fb89f612ecd79f259
commit-date: 2024-03-18
host: aarch64-apple-darwin
release: 1.78.0-beta.1
LLVM version: 18.1.2

**Comment by y21** [comment]
The fix isn't on nightly yet (clippy is only sync'd with upstream every two weeks, I believe the next sync happens on the 21st), so you'll have to wait a bit longer, or you could try to run clippy master locally on your project to confirm that this is fixed (or isn't fixed).

**Comment by ilyagr** [comment]
I see, thanks for letting me know!

Will the fix be backported to 1.78 stable?

I'm not sure I know how to run the master clippy. I thought that was what nightly was for...

**Comment by y21** [comment]
> I'm not sure I know how to run the master clippy.

The easiest way is cloning this repository, cd'ing into it and running `cargo dev lint /path/to/your/project`.

**Comment by ilyagr** [comment]
Thanks! Indeed, that doesn't seem to trigger `clippy::empty_docs`, so I think the fix works.
